src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp

```diff
@@ -717,8 +717,15 @@ ov::pass::RoPEShareCosSin::RoPEShareCosSin() {
         if (!validator) {
             return false;
         }
+
         auto it = pattern_map.find(const_inv_freq);
+        if (it == pattern_map.end()) {
+            return false;
+        }
         auto cur_inv_freq = std::dynamic_pointer_cast<opset1::Constant>(it->second.get_node_shared_ptr());
+        if (!cur_inv_freq) {
+            return false;
+        }
```

Comment:
Can't we just do 

```
auto cur_inv_freq = std::dynamic_pointer_cast<opset1::Constant>(pattern_map[const_inv_freq].get_node_shared_ptr());
if (!cur_inv_freq) {
     return false;
}
```

The reason I haven't done that yet is it is here probably for a reason and the pattern is created using some unusual functionality.